src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp

```diff
@@ -597,12 +613,12 @@ namespace OpenMS
 
   }
 
-  void FeatureFinderIdentificationAlgorithm::createAssayLibrary_(PeptideMap& peptide_map, PeptideRefRTMap& ref_rt_map)
+  void FeatureFinderIdentificationAlgorithm::createAssayLibrary_(const PeptideMap::iterator& begin, const PeptideMap::iterator& end, PeptideRefRTMap& ref_rt_map)
   {
     std::set<String> protein_accessions;
 
-    for (PeptideMap::iterator pm_it = peptide_map.begin();
-         pm_it != peptide_map.end(); ++pm_it)
+    for (auto pm_it = begin;
```

Comment:
not sure if removing all autos here is not detrimental to understanding the code

Reply:
and below

Reply:
I did it because I wanted to make it const, until I saw that we clear the values in the end.
auto would allow for changing it to const iterators later.
And it gets rid of my clang-tidy warnings..